src/UI/App.hs

```diff
@@ -31,8 +36,34 @@ drawUI s =
         ComposeEditor -> drawComposeEditor s
         Help -> drawHelp s
 
+class EventHandler (m :: Mode) where
+  handle :: Proxy m -> AppState -> Event -> T.EventM Name (T.Next AppState)
+  fallback :: Proxy m -> AppState -> Event -> T.EventM Name (T.Next AppState)
```

Comment:
Not sure if I really need the `fallback` here, since I using the `handleEvent` function would work fine for most items.